.github/workflows/cd.yml

```diff
@@ -37,24 +37,54 @@ jobs:
             echo "SERVICE_NAME=${{ vars.AWS_ECS_API_SERVICE_STAGING }}" >> $GITHUB_ENV
           fi
 
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v2
+      - name: Check for existing image with GIT_SHA
+        env:
+          GIT_SHA: ${{ github.sha }}
+          REPO: ${{ vars.AWS_ECR_API_REPO }}
+        run: |
+          image_manifest=$(aws ecr batch-get-image --repository-name $REPO --image-ids imageTag=$GIT_SHA --query 'images[0].imageManifest')
+          echo "IMAGE_MANIFEST=$image_manifest" >> $GITHUB_ENV
+
+      - if: env.IMAGE_MANIFEST == 'null'
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - if: env.IMAGE_MANIFEST == 'null'
+        name: Build, tag, and push docker image to Amazon ECR
+        uses: docker/build-push-action@v5
         with:
-          platforms: arm64
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
```

Comment:
looks like `mode=max` is about caching intermediate (non-exported) Docker layers

do you know if our build has any of those?

Reply:
I don't think so. Went with max just because it could cache more but probably doesn't make a difference for the api image.